src/promnesia/__main__.py

```diff
@@ -269,13 +328,13 @@ def main() -> None:
     ap.add_argument('--name', type=str, default='demo'               , help='Set custom source name')
     add_port_arg(ap)
     ap.add_argument('--no-serve', action='store_const', const=None, dest='port', help='Pass to only index without running server')
-    ap.add_argument('--config', type=Path, required=False            , help='Config to run against. If omitted, will use empty base config')
```

Comment:
Hmm, I think the `test_demo` failure was caused by this, because here the config argument is optional (which means that it would use the temporary directory, see [here](https://github.com/karlicoss/promnesia/blob/65e77cd67740ef391cba73bc9eac200d9a5b6e63/src/promnesia/__main__.py#L110-L121) ).
Whereas in `index` there is `default_config_path()`, this is because under tests we want to avoid running against the actual user config, so it uses a different directory (`promnesia-test`).

I kinda see your point about reusing the arguments though -- in principle indexing is part of `demo` mode, so this makes sense.. Maybe a 'cheap' way to fix it would be to pass `default_config` to `add_index_args`? So index would use `default_config_path()`, and adhoc will pass `None`, and later we can think what would be a 'proper' way to unify them?

Reply:
Completely missed the different values...so i'm either too old to program, or it's the 5 month curfew here in Greece that has taxed heavily on my mental health, or both :-/
Anyway, fixed it, and can't think of another 'proper' way, this seems to be just right, added some documentation that seem appropriate.

Reply:
@ankostis no worries, took me a little to figure out as well even though I wrote the code, so don't think it's the age :) 
(also sorry to hear about the curfew! we don't have that in London, just lockdown and it already sucks, can't imagine when it's also with curfew rules)

Reply:
Noticed that it fails in python<3.8 due to argparse `action='extend'`...i can downgrade it or hope that 3.7 is dying a slow death?

Reply:
Ah, you were 10 seconds ahead of me, I left the comment :) 
3.6 maybe almost dead? But I think 3.7 would be a little premature, many linux/osx installs have it. (even circleci, apparently!)